src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurStandardAccountingExtractCollectorBatchBuilder.java

```diff
@@ -282,44 +321,147 @@ protected String buildTransactionDescription(ConcurStandardAccountingExtractDeta
                 .toString();
     }
 
-    protected void compareLineNonKeyFieldsAgainstEntry(String itemKey,
-            ConcurStandardAccountingExtractDetailLine saeLine, OriginEntryFull originEntry) {
-        if (!StringUtils.equals(originEntry.getDocumentNumber(), buildDocumentNumber(saeLine))) {
-            LOG.warn("Found a documentNumber/reportId mismatch on at least one line for item key: " + itemKey);
-        } else if (!StringUtils.equals(originEntry.getTransactionLedgerEntryDescription(), buildTransactionDescription(saeLine))) {
-            LOG.warn("Found a lastName/firstName/endDate mismatch on at least one line for item key: " + itemKey);
-        }
+    protected void reportCashAdvance(ConcurStandardAccountingExtractDetailLine saeLine) {
+        updateStatisticFromLineData(ConcurStandardAccountingExtractBatchReportData::getCashAdvancesRelatedToExpenseReports, saeLine);
+    }
+
+    protected void reportCorporateCardPayment(ConcurStandardAccountingExtractDetailLine saeLine) {
+        updateStatisticFromLineData(ConcurStandardAccountingExtractBatchReportData::getExpensesPaidOnCorporateCard, saeLine);
+    }
+
+    protected void reportBypassOfLineWithPseudoPaymentCode(ConcurStandardAccountingExtractDetailLine saeLine) {
+        updateStatisticFromLineData(ConcurStandardAccountingExtractBatchReportData::getTransactionsBypassed, saeLine);
+    }
+
+    protected void updateStatisticFromLineData(
+            Function<ConcurStandardAccountingExtractBatchReportData,ConcurBatchReportSummaryItem> summaryItemGetter,
+            ConcurStandardAccountingExtractDetailLine saeLine) {
+        ConcurBatchReportSummaryItem summaryItem = summaryItemGetter.apply(reportData);
+        summaryItem.setRecordCount(summaryItem.getRecordCount() + 1);
+        summaryItem.setDollarAmount(summaryItem.getDollarAmount().add(saeLine.getJournalAmount().abs()));
+    }
+
+    protected void reportPendingClientLine(ConcurStandardAccountingExtractDetailLine saeLine) {
+        ConcurBatchReportMissingObjectCodeItem pendingClientItem = new ConcurBatchReportMissingObjectCodeItem(

```

Comment:
I am asking, do we need this here, we are reporting pending client object codes while processing PDP?  Will we have duplicate error lines in the report?  Is that a problem?